pkg/upgrade/v1_11_0.go

```diff
@@ -0,0 +1,11 @@
+package upgrade
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+func upgrade1_11_0(client client.Client, jaeger v1.Jaeger) (v1.Jaeger, error) {
```

Comment:
These upgrade functions don't appear to be doing anything? So not sure what is going to trigger a redeployment with the new version?

Ideally we need a simple mechanism that can be used in most cases, where no migration is actually required - just a restart with the newer image versions.

Reply:
> So not sure what is going to trigger a redeployment with the new version?

I believe that the update to the `Status` object with the new version will trigger a reconciliation logic, which will end up generating different image specifications for the containers inside the deployments. This image change should trigger the re-deployment of the new deployments. I'll do a manual test to confirm.

Reply:
Just confirmed that this is indeed the case:

1. starting with this PR as the base for the operator, I created a `simplest` CR
1. stopped the operator, `kubectl edit jaeger simplest` and manually changed the Version field to 1.12.0, and changed the Image field for the deployment to 1.12.0 as well (`kubectl edit deployment simplest`
1. started the operator again
1. both the CR and the deployment are upgraded to 1.13.1

Here's the deployment before step 3:
```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    linkerd.io/inject: disabled
    prometheus.io/port: "14269"
    prometheus.io/scrape: "true"
    sidecar.istio.io/inject: "false"
  creationTimestamp: 2019-07-30T13:14:27Z
  generateName: simplest4-856698b8d7-
  labels:
    app: jaeger
    app.kubernetes.io/component: all-in-one
    app.kubernetes.io/instance: simplest4
    app.kubernetes.io/managed-by: jaeger-operator
    app.kubernetes.io/name: simplest4
    app.kubernetes.io/part-of: jaeger
    pod-template-hash: 856698b8d7
  name: simplest4-856698b8d7-72vpp
  namespace: default
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: simplest4-856698b8d7
    uid: cecbf235-dc9c-4cde-ac96-02d95cca6729
  resourceVersion: "4101"
  selfLink: /api/v1/namespaces/default/pods/simplest4-856698b8d7-72vpp
  uid: efb7ef64-1a8a-451b-9379-73880deb35e7
spec:
  containers:
  - args:
    - --sampling.strategies-file=/etc/jaeger/sampling/sampling.json
    env:
    - name: SPAN_STORAGE_TYPE
      value: memory
    - name: COLLECTOR_ZIPKIN_HTTP_PORT
      value: "9411"
    image: jaegertracing/all-in-one:1.12.0
    imagePullPolicy: IfNotPresent
    name: jaeger
    ports:
    - containerPort: 5775
      name: zk-compact-trft
      protocol: UDP
    - containerPort: 5778
      name: config-rest
      protocol: TCP
    - containerPort: 6831
      name: jg-compact-trft
      protocol: UDP
    - containerPort: 6832
      name: jg-binary-trft
      protocol: UDP
    - containerPort: 9411
      name: zipkin
      protocol: TCP
    - containerPort: 14267
      name: c-tchan-trft
      protocol: TCP
    - containerPort: 14268
      name: c-binary-trft
      protocol: TCP
    - containerPort: 16686
      name: query
      protocol: TCP
    - containerPort: 14269
      name: admin-http
      protocol: TCP
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /
        port: 14269
        scheme: HTTP
      initialDelaySeconds: 1
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/jaeger/sampling
      name: simplest4-sampling-configuration-volume
      readOnly: true
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: simplest4-token-zd559
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: minikube
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: simplest4
  serviceAccountName: simplest4
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - configMap:
      defaultMode: 420
      items:
      - key: sampling
        path: sampling.json
      name: simplest4-sampling-configuration
    name: simplest4-sampling-configuration-volume
  - name: simplest4-token-zd559
    secret:
      defaultMode: 420
      secretName: simplest4-token-zd559
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:14:28Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:14:40Z
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:14:40Z
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:14:27Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://e4d47c3c47205fafe5aca9a5a9941ce89b7d0125b5942ceb4d0f7f6c37083d4b
    image: jaegertracing/all-in-one:1.12.0
    imageID: docker-pullable://jaegertracing/all-in-one@sha256:51b901b653f4a4ca5dd50be9133c08dacf2d3eb092e507c213e7955a0e132297
    lastState: {}
    name: jaeger
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2019-07-30T13:14:33Z
  hostIP: 192.168.122.73
  phase: Running
  podIP: 172.17.0.9
  qosClass: BestEffort
  startTime: 2019-07-30T13:14:28Z
```

And here is the same deployment, after the step 3 has finished:

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    linkerd.io/inject: disabled
    prometheus.io/port: "14269"
    prometheus.io/scrape: "true"
    sidecar.istio.io/inject: "false"
  creationTimestamp: 2019-07-30T13:17:55Z
  generateName: simplest4-6f9c454bcf-
  labels:
    app: jaeger
    app.kubernetes.io/component: all-in-one
    app.kubernetes.io/instance: simplest4
    app.kubernetes.io/managed-by: jaeger-operator
    app.kubernetes.io/name: simplest4
    app.kubernetes.io/part-of: jaeger
    pod-template-hash: 6f9c454bcf
  name: simplest4-6f9c454bcf-fjfnx
  namespace: default
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: simplest4-6f9c454bcf
    uid: d47ce831-8d36-42ea-9a04-31972d775dae
  resourceVersion: "4433"
  selfLink: /api/v1/namespaces/default/pods/simplest4-6f9c454bcf-fjfnx
  uid: 9bc36ec6-6cb8-4850-86a6-ae85a0ed46bb
spec:
  containers:
  - args:
    - --sampling.strategies-file=/etc/jaeger/sampling/sampling.json
    env:
    - name: SPAN_STORAGE_TYPE
      value: memory
    - name: COLLECTOR_ZIPKIN_HTTP_PORT
      value: "9411"
    image: jaegertracing/all-in-one:1.13.1
    imagePullPolicy: IfNotPresent
    name: jaeger
    ports:
    - containerPort: 5775
      name: zk-compact-trft
      protocol: UDP
    - containerPort: 5778
      name: config-rest
      protocol: TCP
    - containerPort: 6831
      name: jg-compact-trft
      protocol: UDP
    - containerPort: 6832
      name: jg-binary-trft
      protocol: UDP
    - containerPort: 9411
      name: zipkin
      protocol: TCP
    - containerPort: 14267
      name: c-tchan-trft
      protocol: TCP
    - containerPort: 14268
      name: c-binary-trft
      protocol: TCP
    - containerPort: 16686
      name: query
      protocol: TCP
    - containerPort: 14269
      name: admin-http
      protocol: TCP
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /
        port: 14269
        scheme: HTTP
      initialDelaySeconds: 1
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/jaeger/sampling
      name: simplest4-sampling-configuration-volume
      readOnly: true
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: simplest4-token-zd559
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: minikube
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: simplest4
  serviceAccountName: simplest4
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - configMap:
      defaultMode: 420
      items:
      - key: sampling
        path: sampling.json
      name: simplest4-sampling-configuration
    name: simplest4-sampling-configuration-volume
  - name: simplest4-token-zd559
    secret:
      defaultMode: 420
      secretName: simplest4-token-zd559
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:17:55Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:18:06Z
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:18:06Z
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: 2019-07-30T13:17:55Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://d806fc5afa59907bccdb43d98e6e495ca40e1cc6ee80360147149a5b9dac7318
    image: jaegertracing/all-in-one:1.13.1
    imageID: docker-pullable://jaegertracing/all-in-one@sha256:4a312d474cf72cb855237a52221c34796ac3047dc76e18145f7846859eb4a2cf
    lastState: {}
    name: jaeger
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2019-07-30T13:17:56Z
  hostIP: 192.168.122.73
  phase: Running
  podIP: 172.17.0.8
  qosClass: BestEffort
  startTime: 2019-07-30T13:17:55Z
```